applications/nrf_desktop/src/modules/power_manager.c

```diff
@@ -49,24 +49,41 @@ static struct k_delayed_work power_down_trigger;
 static struct k_delayed_work error_trigger;
 static atomic_t power_down_count;
 static unsigned int connection_count;
-static bool usb_connected;
+static enum usb_state usb_state;
 
 
+static void system_off(void)
+{
+	if (power_state == POWER_STATE_ERROR_SUSPENDED) {
+		power_state = POWER_STATE_ERROR_OFF;
+	} else {
+		power_state = POWER_STATE_OFF;
+	}
+
+	LOG_WRN("System turned off");
+	LOG_PANIC();
+
+	sys_pm_force_power_state(SYS_POWER_STATE_DEEP_SLEEP_1);
+}
+
 static void power_down(struct k_work *work)
 {
-	__ASSERT_NO_MSG(!usb_connected);
-	__ASSERT_NO_MSG(power_state == POWER_STATE_IDLE);
+	__ASSERT_NO_MSG(usb_state != USB_STATE_POWERED);
 
 	atomic_val_t cnt = atomic_add(&power_down_count, POWER_DOWN_CHECK_MS);
 
 	if (cnt >= POWER_DOWN_TIMEOUT_MS) {
-		LOG_INF("System power down");
+		if (power_state == POWER_STATE_SUSPENDED) {
```

Comment:
It can. This occurs when the device first gets USB suspended enters standby and then gets USB disconnected. Power down workqueue is started to bring the board down. It is not done immediately as USB disconnect happens right after USB is woken up - at this time we cannot turn the system off. This is why I left the usual timeout to happen.

Reply:
Ok, then you need to verify if the device is connected over Bluetooth too

Reply:
We may also consider separate work that could be used to perform transition from suspended state to system turned off (to separate the functionalities and keep common code related to `system_power_off` in one place)

Reply:
added more checks